src/image/image.cpp

```diff
@@ -1,62 +1,82 @@
 #include "image.h"
 
-Persistent<FunctionTemplate> LwipImage::constructor;
+Nan::Persistent<FunctionTemplate> LwipImage::constructor;
 
 void LwipImage::Init(Handle<Object> exports) {
     // Prepare constructor template
-    Local<FunctionTemplate> tpl = NanNew<FunctionTemplate>(New);
+    Local<FunctionTemplate> tpl = Nan::New<FunctionTemplate>(New);
     tpl->InstanceTemplate()->SetInternalFieldCount(1);
-    tpl->SetClassName(NanNew("LwipImage"));
-    NODE_SET_PROTOTYPE_METHOD(tpl, "width", width);
-    NODE_SET_PROTOTYPE_METHOD(tpl, "height", height);
-    NODE_SET_PROTOTYPE_METHOD(tpl, "getPixel", getPixel);
-    NODE_SET_PROTOTYPE_METHOD(tpl, "buffer", buffer);
-    NODE_SET_PROTOTYPE_METHOD(tpl, "resize", resize);
-    NODE_SET_PROTOTYPE_METHOD(tpl, "rotate", rotate);
-    NODE_SET_PROTOTYPE_METHOD(tpl, "blur", blur);
-    NODE_SET_PROTOTYPE_METHOD(tpl, "crop", crop);
-    NODE_SET_PROTOTYPE_METHOD(tpl, "mirror", mirror);
-    NODE_SET_PROTOTYPE_METHOD(tpl, "pad", pad);
-    NODE_SET_PROTOTYPE_METHOD(tpl, "sharpen", sharpen);
-    NODE_SET_PROTOTYPE_METHOD(tpl, "hslaAdj", hslaAdj);
-    NODE_SET_PROTOTYPE_METHOD(tpl, "opacify", opacify);
-    NODE_SET_PROTOTYPE_METHOD(tpl, "paste", paste);
-    NODE_SET_PROTOTYPE_METHOD(tpl, "setPixel", setPixel);
-    NanAssignPersistent(constructor, tpl);
+    tpl->SetClassName(Nan::New("LwipImage").ToLocalChecked());
+    Nan::SetPrototypeMethod(tpl, "width", width);
+    Nan::SetPrototypeMethod(tpl, "height", height);
+    Nan::SetPrototypeMethod(tpl, "getPixel", getPixel);
+    Nan::SetPrototypeMethod(tpl, "buffer", buffer);
+    Nan::SetPrototypeMethod(tpl, "resize", resize);
+    Nan::SetPrototypeMethod(tpl, "rotate", rotate);
+    Nan::SetPrototypeMethod(tpl, "blur", blur);
+    Nan::SetPrototypeMethod(tpl, "crop", crop);
+    Nan::SetPrototypeMethod(tpl, "mirror", mirror);
+    Nan::SetPrototypeMethod(tpl, "pad", pad);
+    Nan::SetPrototypeMethod(tpl, "sharpen", sharpen);
+    Nan::SetPrototypeMethod(tpl, "hslaAdj", hslaAdj);
+    Nan::SetPrototypeMethod(tpl, "opacify", opacify);
+    Nan::SetPrototypeMethod(tpl, "paste", paste);
+    Nan::SetPrototypeMethod(tpl, "setPixel", setPixel);
+    constructor.Reset(tpl);
     exports->Set(
-        NanNew("LwipImage"),
-        NanNew<FunctionTemplate>(constructor)->GetFunction()
+        Nan::New("LwipImage").ToLocalChecked(),
+        Nan::New<FunctionTemplate>(constructor)->GetFunction()
     );
 }
 
-LwipImage::LwipImage(unsigned char * data, size_t width, size_t height) {
+LwipImage::LwipImage(unsigned char * data, size_t width, size_t height, int channels) {
     // TODO: CImg constructor may throw an exception. handle it in LwipImage::New.
-    _cimg = new CImg<unsigned char>(data, width, height, 1, N_CHANNELS, false);
+    _cimg = new CImg<unsigned char>(data, width, height, 1, channels, false);
```

Comment:
Passing to CImg the correct number of channels is indeed necessary. But the intention was that all buffers have 4 channels. If some buffers are passed which do not have 4 channels, then it's a bug in the code that passed the buffer.
I want to trace the area of the code which passed those invalid buffers and fix it there.
The reason I want all buffer to be 4 channels is that the encoders expect 4 channels.